dohproxy/client_protocol.py

```diff
@@ -50,10 +60,20 @@ async def setup_client(self):
     def connection_made(self, transport):
         self.transport = transport
 
+    def is_passthrough_query(self, dnsq):
+        for question in dnsq.question:
```

Comment:
I would be tempted to only support single question..... but I think ATM, the proxy pretty much support multiple questions as it does not care about the content really.
I see this being a potential surface of abuse if `passthrough_domains` grows in size but probably fine depending on the use-case and where this is used. Maybe a configurable option to limit how many questions are check would help, with default to 1 being sensible.